tests/test_extract.py

```diff
@@ -1,49 +1,141 @@
+    mock_df = MockAsDF()
+    with patch.object(syn, "tableQuery", return_value=mock_df) as patch_syn_tablequery:
+        df = extract.read_table_into_df(table_id="syn11111111", syn=syn)
+        patch_syn_tablequery.assert_called_once_with("select * from syn11111111")
+        assert isinstance(df, pd.DataFrame)
+        assert df.equals(mock_df.asDataFrame())
+
+
+def test_read_feather_into_df_if_not_feather():
+    with pytest.raises(ValueError, match="Please make sure *"):
+        extract.read_feather_into_df(feather_path="./tests/test_assets/test_file.csv")
+
+
+def test_read_feather_into_df():
+    with patch.object(
+        pd, "read_feather", return_value=pd.DataFrame()
+    ) as patch_pd_read_feather:
+        df = extract.read_feather_into_df(
+            feather_path="./tests/test_assets/test_file.feather"
+        )
+        patch_pd_read_feather.assert_called_once_with(
+            "./tests/test_assets/test_file.feather"
+        )
+        assert isinstance(df, pd.DataFrame)
+
 
-def test_get_entity_as_df():
-    assert type(extract.get_entity_as_df(syn_id="syn25838562", format="csv")) is DataFrame
-    with pytest.raises(SystemExit) as err:
-        extract.get_entity_as_df(syn_id="syn25838562", format="tsv")
-    assert err.type == SystemExit
-    assert err.value.code == 9
+def test_read_json_into_df_if_not_json():
+    with pytest.raises(ValueError, match="Please make sure *"):
+        extract.read_json_into_df(json_path="./tests/test_assets/test_file.csv")
 
-    assert type(extract.get_entity_as_df(syn_id="syn25838563", format="tsv")) is DataFrame
-    with pytest.raises(SystemExit) as err:
-        extract.get_entity_as_df(syn_id="syn25838563", format="csv")
-    assert err.type == SystemExit
-    assert err.value.code == 9
 
-    assert type(extract.get_entity_as_df(syn_id="syn25838546", format="table")) is DataFrame
-    with pytest.raises(SystemExit) as err:
-        extract.get_entity_as_df(syn_id="syn25838563", format="table")
-    assert err.type == SystemExit
-    assert err.value.code == 1
+def test_read_json_into_df():
+    with patch.object(
+        pd, "read_json", return_value=pd.DataFrame()
+    ) as patch_pd_read_json:
+        df = extract.read_json_into_df(json_path="./tests/test_assets/test_file.json")
+        patch_pd_read_json.assert_called_once_with(
+            "./tests/test_assets/test_file.json", orient="records"
+        )
+        assert isinstance(df, pd.DataFrame)
+
+
+# test if utils._login_to_synapse is called when syn=None
+def test_get_entity_as_df_syn_is_none(syn):
+    with patch.object(
+        utils, "_login_to_synapse", return_value=syn
+    ) as patch_login_to_synapse:
+        extract.get_entity_as_df(syn_id="syn11111111", format="table", syn=None)
+        patch_login_to_synapse.assert_called_once()
+
+
+# dummy synapse entity needed to supply entity.path to read_csv_into_df for next test
+ENTITY = synapseclient.File("fake/path.csv", parent="syn1111111")
```

Comment:
Nit: Put the ENTITY within the function since it doesn't seem to be used else where.